### PR TITLE
feat: in code config

### PR DESCRIPTION
**Pull request by Jakub-Vacek**
This PR enables passing IConfig variables through SuperfaceClient constructor. These values are merged with config loaded from environment. Value passed in code has higher priority. Properties like `debug` will be added in separate PR.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have updated the documentation accordingly. If you are changing **code related to user secrets** you need to really make sure that [security documentation](https://github.com/superfaceai/one-sdk-js/blob/main/SECURITY.md) is correct.
- [x] I have read the [CONTRIBUTING](https://github.com/superfaceai/one-sdk-js/blob/main/CONTRIBUTING.md) document.
- [x] I haven't repeated the code. (DRY)
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/core/config/config.ts [hunk 1]
```diff
@@ -171,27 +183,121 @@ export class Config implements IConfig {
...
+
 export function loadConfigFromEnv(
   environment: IEnvironment,
   fileSystem: FSPath,
   logger?: ILogger
 ): Config {
   const env = {
     superfaceApiUrl: getSuperfaceApiUrl(environment),
-    sdkAuthToken: getSdkAuthToken(environment),
+    sdkAuthToken: getSdkAuthToken(environment, logger?.log(DEBUG_NAMESPACE)),
```
Comment (lukas-valenta) [inline src/core/config/config.ts:272]: nitpick: you can store the log function in a variable

File: src/core/config/config.ts [hunk 2]
```diff
@@ -171,27 +183,121 @@ export class Config implements IConfig {
...
+  logger?: ILogger
+): Config {
+  const env = {
+    superfaceApiUrl: ensureValidUrl(config.superfaceApiUrl),
+    sdkAuthToken: ensureValidSdkToken(
+      config.sdkAuthToken,
+      'sdkAuthToken',
+      logger?.log(DEBUG_NAMESPACE)
+    ),
+    // TODO: Check if it is path?
```
Comment (lukas-valenta) [inline src/core/config/config.ts:232]: :+1:

File: src/core/config/config.ts [hunk 3]
```diff
@@ -171,27 +183,121 @@ export class Config implements IConfig {
...
+    superfacePath: config.superfacePath,
+    superfaceCacheTimeout: ensurePositiveInteger(
+      config.superfaceCacheTimeout,
+      'superfaceCacheTimeout',
+      logger?.log(DEBUG_NAMESPACE)
+    ),
+    // TODO: check if min is less than max?
```
Comment (lukas-valenta) [inline src/core/config/config.ts:239]: I'd do that in reporter, not here

File: src/core/config/config.ts [hunk 4]
```diff
@@ -171,27 +183,121 @@ export class Config implements IConfig {
...
+      'metricDebounceTimeMin',
+      logger?.log(DEBUG_NAMESPACE)
+    ),
+    metricDebounceTimeMax: ensurePositiveInteger(
+      config.metricDebounceTimeMax,
+      'metricDebounceTimeMax',
+      logger?.log(DEBUG_NAMESPACE)
+    ),
+    disableReporting: config.disableReporting,
+    // TODO: Check if it is path?
```
Comment (lukas-valenta) [inline src/core/config/config.ts:251]: :+1:

File: src/node/client/client.ts [hunk 5]
```diff
@@ -100,14 +122,22 @@ export abstract class SuperfaceClientBase {
   protected readonly fetchInstance: IFetch & Interceptable & AuthCache;
   protected readonly logger?: ILogger;
 
-  constructor(options?: { superJson?: SuperJson | SuperJsonDocument }) {
+  constructor(
+    // TODO: for now omit cachePath( does not have env variable, conflicts with our plans for the API)?
```
Comment (lukas-valenta) [inline src/node/client/client.ts:126]: :+1:

File: src/core/config/config.ts (hunk 2 above)
Comment (Jakub-Vacek) [reply src/core/config/config.ts:233]: @lukas-valenta  FS isAbsolute can be used to check if path is absolute. For relative I would probably had to use regex. I think we should leave it for now.

**Review by lukas-valenta — APPROVED** [review]

**Pull request merged**